src/sys/unix.rs

```diff
@@ -1597,6 +1597,93 @@ impl crate::Socket {
         ))
         .map(|_| sbytes as usize)
     }
+
+    /// Set the value of the `TCP_USER_TIMEOUT` option on this socket.
+    ///
+    /// If set, this specifies the maximum amount of time that transmitted data may remain
+    /// unacknowledged or buffered data may remain untransmitted before TCP will forcibly close the
+    /// corresponding connection.
+    #[cfg(all(
+        feature = "all",
+        any(
+            target_os = "android",
+            target_os = "emscripten",
+            target_os = "fuchsia",
+            target_os = "l4re",
+            target_os = "linux",
+        )
+    ))]
+    #[cfg_attr(
+        docsrs,
+        doc(cfg(all(
+            feature = "all",
+            any(
+                target_os = "android",
+                target_os = "emscripten",
+                target_os = "fuchsia",
+                target_os = "l4re",
+                target_os = "linux",
+            )
+        )))
+    )]
+    pub fn set_tcp_user_timeout(&self, duration: Option<Duration>) -> io::Result<()> {
+        use std::convert::TryInto;
+        fn into_millis(duration: Option<Duration>) -> io::Result<c_int> {
+            duration
+                .map_or(0, |d| d.as_millis())
+                .try_into()
+                .map_err(|e| io::Error::new(io::ErrorKind::Other, e))
+        }
```

Comment:
Can you remove the function and replace it with the following.

```rust
        let timeout = timeout
            .map(|to| cmp::min(to.as_millis(), libc::c_uint::MAX as u128) as libc::c_uint)
            .unwrap_or(0);
```

Reply:
https://github.com/rust-lang/socket2/pull/241/commits/df14e0d8c5dd6ddc4056e0c4cad1509ea49993c8